src/client/visualizations/bar-chart/improved-bar-chart/utils/bar-chart-model.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Timezone } from "chronoshift";
+import { List, OrderedMap } from "immutable";
+import { Dataset, Datum } from "plywood";
+import { NORMAL_COLORS } from "../../../../../common/models/colors/colors";
+import { Dimension } from "../../../../../common/models/dimension/dimension";
+import { Essence } from "../../../../../common/models/essence/essence";
+import { ConcreteSeries } from "../../../../../common/models/series/concrete-series";
+import { Split } from "../../../../../common/models/split/split";
+import { Omit } from "../../../../../common/utils/functional/functional";
+import { selectFirstSplitDatums } from "../../../../utils/dataset/selectors/selectors";
+import { getContinuousSplit, getNominalDimension, getNominalSplit, hasNominalSplit } from "../../../line-chart/utils/splits";
```

Comment:
This is the only place we import utils from line-chart. I think we can leave them here for now and see if we add non-continuous dimensions nomenclature would change.

Reply:
Do you know any tool for generating typescript packages/imports dependency matrix. Like this:
https://www.mscharhag.com/files/2015/jarchitect-depmatrix.png

We could make a broader effort to break all unwanted dependencies (like cyclic ones).